ci/code_checks.sh

```diff
@@ -14,6 +14,8 @@
 #   $ ./ci/code_checks.sh single-docs   # check single-page docs build warning-free
 #   $ ./ci/code_checks.sh notebooks     # check execution of documentation notebooks
 
+set -euo pipefail
```

Comment:
Does using `e` here mean the script will exit on the first failure? It can be useful to see all failures.

Reply:
Correct. Yeah if one runs this locally I can see it being useful seeing all the check failures. In the Ci we call this script with just one option at a time. I'll just remove `e`

Reply:
And removing `e` won't cover up any failure on the CI?

Reply:
It shouldn't, no. Mostly likely if an intermediate step fails the whole step will fail. I'm thinking of using `set -e` in the CI soon anyways